opm/input/eclipse/Generator/KeywordGenerator.cpp

```diff
@@ -17,252 +17,276 @@
   along with OPM.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-#include <fstream>
-#include <filesystem>
-#include <iostream>
-#include <sstream>
-#include <cctype>
-#include <fmt/format.h>
-
-#include <opm/json/JsonObject.hpp>
 #include <opm/input/eclipse/Generator/KeywordGenerator.hpp>
+
 #include <opm/input/eclipse/Generator/KeywordLoader.hpp>
+
+#include <opm/json/JsonObject.hpp>
+
 #include <opm/input/eclipse/Parser/ParserKeyword.hpp>
 
+#include <cctype>
+#include <filesystem>
+#include <fstream>
+#include <map>
+#include <sstream>
+#include <string>
+
+#include <fmt/format.h>
 
 namespace {
 
-const std::string sourceHeader = R"(
+    void updateFile(const std::string& newContent, const std::string& filename)
+    {
+        Opm::KeywordGenerator::ensurePath(filename);
+
+        std::ofstream { filename } << newContent;
+    }
+
+    void write_file(const std::string& content,
+                    const std::string& file,
+                    const bool         verbose,
+                    const std::string& desc)
+    {
+        updateFile(content, file);
+
+        if (verbose) {
+            fmt::print("Updated {} file written to {}\n", desc, file);
+        }
+    }
+
+    void write_file(const std::stringstream& stream,
+                    const std::string&       file,
+                    const bool               verbose,
+                    const std::string&       desc = "source")
```

Comment:
here as well?

Reply:
> here as well?

:persevere: 

Thanks&ndash;I should have been more thorough before.  I've pushed an update that I **believe** fixes that one as well...